napari/view_layers.py

```diff
@@ -139,39 +145,153 @@ def _make_viewer_then(add_method: str, args, kwargs) -> Viewer:
 
 @_merge_layer_viewer_sigs_docs
 def view_image(*args, **kwargs):
-    return _make_viewer_then('add_image', args, kwargs)
+    return _make_viewer_then('add_image', args, kwargs)[0]
 
 
 @_merge_layer_viewer_sigs_docs
 def view_labels(*args, **kwargs):
-    return _make_viewer_then('add_labels', args, kwargs)
+    return _make_viewer_then('add_labels', args, kwargs)[0]
 
 
 @_merge_layer_viewer_sigs_docs
 def view_points(*args, **kwargs):
-    return _make_viewer_then('add_points', args, kwargs)
+    return _make_viewer_then('add_points', args, kwargs)[0]
 
 
 @_merge_layer_viewer_sigs_docs
 def view_shapes(*args, **kwargs):
-    return _make_viewer_then('add_shapes', args, kwargs)
+    return _make_viewer_then('add_shapes', args, kwargs)[0]
 
 
 @_merge_layer_viewer_sigs_docs
 def view_surface(*args, **kwargs):
-    return _make_viewer_then('add_surface', args, kwargs)
+    return _make_viewer_then('add_surface', args, kwargs)[0]
 
 
 @_merge_layer_viewer_sigs_docs
 def view_tracks(*args, **kwargs):
-    return _make_viewer_then('add_tracks', args, kwargs)
+    return _make_viewer_then('add_tracks', args, kwargs)[0]
 
 
 @_merge_layer_viewer_sigs_docs
 def view_vectors(*args, **kwargs):
-    return _make_viewer_then('add_vectors', args, kwargs)
+    return _make_viewer_then('add_vectors', args, kwargs)[0]
 
 
 @_merge_layer_viewer_sigs_docs
 def view_path(*args, **kwargs):
-    return _make_viewer_then('open', args, kwargs)
+    return _make_viewer_then('open', args, kwargs)[0]
+
+
+def imshow(*args, viewer=None, **kwargs):
```

Comment:
I'm afraid kwargs here should also be laboriously listed out... as we've done for `add_image` ... and it should also include the `viewer.__init__ ` stuff.  Essentially, it needs to manually do what `_merge_layer_viewer_sigs_docs` was doing (which was to build _both_ the signature and the docs.

```python
    def imshow(
        data,
        *,
        channel_axis=None,
        rgb=None,
        colormap=None,
        contrast_limits=None,
        gamma=1,
        interpolation2d='nearest',
        interpolation3d='linear',
        ...
```

using **kwargs one loses all the tab-autocompletion in an interactive terminal or in an IDE.  this is course is super annoying, and it also leads to the possibility that these defaults go out of sync with `add_image`... (and this is why we've autogenerated these in the past).  so, we might also want to add back the test to make sure that all the parameters match.  You can see an old test we used to use here:  https://github.com/napari/napari/blob/871825e9ca9f570575e0681132d638e03cb39708/napari/_tests/test_viewer_layer_parity.py

While I do think the signature should be updated here, I'll let others weigh in on whether to bring back those old tests (which were always a bit annoying TBH).  

Also, @jni should be signing off on this PR as well since it was his proposal

Reply:
@jni, @alisterburt this was not yet addressed. Should we tackle it before merging?

Reply:
Yes absolutely - I'm afk until at least tomorrow but will do then if not done before!